File: src/main/java/work/lclpnet/ap2/impl/game/data/entry/ScoreTimeDataEntry.java
package work.lclpnet.ap2.impl.game.data.entry;

import org.jetbrains.annotations.Nullable;
import work.lclpnet.ap2.api.game.data.DataEntry;
import work.lclpnet.ap2.api.game.data.SubjectRef;
import work.lclpnet.kibu.translate.Translations;
import work.lclpnet.kibu.translate.text.TranslatedText;

public record ScoreTimeDataEntry<Ref extends SubjectRef>(Ref subject, int score, int ranking) implements DataEntry<Ref>, ScoreView {

    @Override
    public @Nullable TranslatedText toText(Translations translationService) {
        return translationService.translateText("ap2.score.points_timed", score, ranking);
    }

    @Override
    public boolean scoreEquals(DataEntry<Ref> _other) {
        if ((_other instanceof ScoreTimeDataEntry<Ref> other)) {
            return score == other.score && ranking == other.ranking;
        }

        return false;
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/data/entry/ScoreTimeDataEntry.kt
package work.lclpnet.ap2.game.data.entry

import work.lclpnet.ap2.game.data.DataEntry
import work.lclpnet.ap2.game.data.SubjectRef
import work.lclpnet.kibu.translate.Translations
import work.lclpnet.kibu.translate.text.TranslatedText

@JvmRecord
data class ScoreTimeDataEntry<Ref : SubjectRef>(
    override val subject: Ref,
    override val score: Int,
    val keyOverride: String?,
    val ranking: Int
) : DataEntry<Ref>, ScoreView {

    override fun toText(translationService: Translations): TranslatedText {
        val key = keyOverride ?: "ap2.score.points_timed"

        return translationService.translateText(key, score, ranking)
    }

    override fun scoreEquals(other: DataEntry<Ref>): Boolean {
        if ((other is ScoreTimeDataEntry<Ref>)) {
            return score == other.score && ranking == other.ranking
        }

        return false
    }
}
